rflx/pyrflx/typevalue.py

```diff
@@ -260,3 +534,311 @@ def accepted_type(self) -> type:
+                self.set(field_name, field_bits)
+
+            return last_pos_in_bitstr, current_pos_in_bitstring
+
+        while current_field_name != FINAL.name and (
+            current_field_first_in_bitstr + current_field_length
+        ) <= len(value):
+
+            current_field = self._fields[current_field_name]
+
+            if isinstance(current_field.typeval, OpaqueValue) and not self._has_length(
+                current_field_name
+            ):
+
+                (
+                    last_field_first_in_bitstr,
+                    current_field_first_in_bitstr,
+                ) = set_field_without_length(current_field_name, current_field)
+
+            elif self._has_length(current_field_name):
+                current_field_length = self._get_length(current_field_name).value
+                try:
+                    (
+                        last_field_first_in_bitstr,
+                        current_field_first_in_bitstr,
+                    ) = set_field_with_length(current_field_name, current_field_length)
+                except IndexError:
+                    raise IndexError(
+                        f"Bitstring representing the message is too short - "
+                        f"stopped while parsing field: {current_field_name}"
+                    )
+            else:
+                if current_field_length == 0:
+                    raise ValueError("Field cannot have a length of zero")
+                if not self._has_length(current_field_name):
+                    raise ValueError("Field does not have a length and is not Opaque")
+
+            current_field_name = self._next_field(current_field_name)
+
+    def set(self, fld: str, value: Union[bytes, int, str, Sequence[TypeValue], Bitstring]) -> None:
+
+        if fld in self.accessible_fields:
+
+            field = self._fields[fld]
+            field.first = self._get_first(fld)
+            if isinstance(value, Bitstring):
+                if isinstance(field.typeval, CompositeValue) and self._has_length(fld):
+                    field.typeval.set_expected_size(self._get_length(fld))
+                field.typeval.parse(value)
+            elif isinstance(value, field.typeval.accepted_type):
+                if isinstance(field.typeval, CompositeValue) and self._has_length(fld):
+                    field.typeval.set_expected_size(self._get_length(fld))
+                field.typeval.assign(value)
+            else:
+                raise TypeError(
+                    f"cannot assign different types: {field.typeval.accepted_type.__name__}"
+                    f" != {type(value).__name__}"
+                )
+
+        else:
+            raise KeyError(f"cannot access field {fld}")
+
+        if all([self.__simplified(o.condition) == FALSE for o in self._model.outgoing(Field(fld))]):
+            self._fields[fld].typeval.clear()
+            raise ValueError("value does not fulfill field condition")
```

Comment:
This error message is not very helpful when the `parse` method was used: Which value? Which field?

Reply:
Fixed by a02992f